### ISSUE TITLE
[Feature request] <Prevent pinned tabs from being closed>

### ISSUE THREAD

**Issue by bill-coyle** [notepad-plus-plus/notepad-plus-plus#15943]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Issue

Current tab pinning moved tabs to leftmost position, but it still can be closed while it is pinned.

### Describe the solution you'd like.

In addition to moving to the left, also prevent the tab from being closed.

### Debug Information

```shell
Notepad++ v8.7.4   (64-bit)
Build time : Dec  4 2024 - 23:50:05
Path : C:\Programs\Text Editors\X-Notepad++\notepad++.exe
Command Line : "C:\[Backup]\Dropbox\NICS\_BD\Unit 04\4.5b - Christ's Kenosis.txt"
Admin mode : ON
Local Conf mode : ON
Cloud Config : OFF
Periodic Backup : OFF
Placeholders : OFF
DirectWrite : ON
Multi-instance Mode : multiInstOnSession
File Status Auto-Detection : cdEnabledOld (for all opened files/tabs) + cdAutoUpdate + cdGo2end
Dark Mode : ON
OS Name : Windows 10 Pro (64-bit)
OS Version : 22H2
OS Build : 19045.5131
Current ANSI codepage : 1252
Plugins : 
    AutoSave (1.6.1)
    ComparePlugin (2.0.2)
    DSpellCheck (1.4.24)
    HexEditor (0.9.12)
    MarkdownViewerPlusPlus (0.8.2)
    mimeTools (3.1)
    NppConverter (4.6)
    NppExport (0.4)
    NppMarkdownPanel (0.6.1)
    NppSnippets (1.7)
    SessionMgr (1.4.4)
```

**Comment by alankilborn** [comment]
Should "Close" (single) UI elements/commands be disabled/greyed when the active tab is pinned?
E.g. _File_ > _Close_
E.g. tab-right-click _Close_
E.g. red `X` on tabs (this could just be made invisible)

"Close" (multiple) UI elements/commands should just proceed as normal, but ignoring any pinned tabs?
E.g. _File_ > _Close All_
E.g. _File_ > _Close All to the Right_

**Comment by bill-coyle** [comment]
Yes, in both cases.

**Comment by phpmoli** [comment]
I need this so much, i have files with 1800000 lines opened, and bookmarked, and when i accidentally close the file and then reopen, the bookmarks are not restored and i lose hours of work.

**Comment by alankilborn** [comment]
It does make sense that pinned tabs should be immune to normal closure mechanisms, i.e., you should have to unpin them before closing is allows.  If that is not to be, then there should at least be a confirmation dialog that occurs for this case.

**Comment by phpmoli** [comment]
also, please be immune to all close procedures, not just the menu ones, but ctrl+W, middle mouse click, etc. Basically check just before closing if it is pinned, and if true, do nothing

**Comment by NovaZek** [comment]
Yeah, pinning them makes no sense, if you can actually remove them with any kind of combination and prompt (alt+w, middle click, etc.). This needs an additional menu point at the preferences, setting it "persistent" or something like that.
Pinning usually is used not only for grouping important things to the top most area (in this case left alignment), but also making them not closable by accident or intention (like it has been always the case at the Browsers like Chrome, Firefox, Vivaldi, you name it).
